compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -734,6 +734,15 @@ pub struct TraitPredicate<'tcx> {
 pub type PolyTraitPredicate<'tcx> = ty::Binder<'tcx, TraitPredicate<'tcx>>;
 
 impl<'tcx> TraitPredicate<'tcx> {
+    pub fn remap_constness(&mut self, tcx: TyCtxt<'tcx>, param_env: &mut ParamEnv<'tcx>) {
+        if unlikely!(Some(self.trait_ref.def_id) == tcx.lang_items().drop_trait()) {
```

Comment:
It seems kind of weird to be special-casing `Drop` here. Are there any other traits that need this kind of special handling? Are you sure that this method is called in all of the places where it needs to be?

Reply:
1. No because we only need to special case `~const Drop` bounds.
2. Yes because I have audited all usages of a method I added made for exactly this purpose. Remapping `T: ~const SomeTrait` to `T: SomeTrait` when not in a const context. 

See also https://rust-lang.zulipchat.com/#narrow/stream/146212-t-compiler.2Fconst-eval/topic/~const.20Drop which would be a better solution